src/services/company.js

```diff
@@ -52,7 +57,50 @@ class CompanyService {
      * @param {*} attributes object containing the attributes to change in company
      */
     changeAttributes(company_id, attributes) {
-        return Company.findOneAndUpdate({ _id: company_id }, attributes);
+        return Company.findOneAndUpdate(
+            { _id: company_id },
+            attributes,
+            { new: true, omitUndefined: true });
+    }
+
+    async disable(company_id) {
+        const company = this.changeAttributes(company_id, { isDisabled: true });
+
+        await Offer.updateMany(
+            { owner: company_id },
+            {
+                isHidden: true,
+                hiddenReason: OfferConstants.HiddenOfferReasons.COMPANY_DISABLED,
+            },
+            { new: true },
+            (err) => {
+                if (err) {
+                    throw err;
+                }
+            }
```

Comment:
After looking at both #155 and this PR I think it's best to wait for #155 , since the logic is the same, only changing the reason to hide offers. What do you think ?

Reply:
#155 is now merged